Assets/Mirror/Components/NetworkAnimator.cs

```diff
@@ -377,7 +392,29 @@ public void SetTrigger(int hash)
             }
         }
 
+        /// <summary>
+        /// Causes an animation trigger to be reset for a networked object.
+        /// </summary>
+        /// <param name="hash">Hash id of trigger (from the Animator).</param>
+        public void ResetTrigger(int hash)
+        {
+            if (hasAuthority && clientAuthority)
```

Comment:
The logic seems right,  but it is a bit hard to understand.

if clientAuthority,   then resetTrigger should only be called by the controlling client.  So how about we print a warning if that is not the case?  a comment would also be useful.

if not client authority,  then this should only be called on the server,  else,  perhaps we should print a warning?   also a comment to explain

So for example:

```
if (clientAuthority) 
{
      // the animator should be controlled by the client
     if (!isClient) 
     {
         Debug.LogWarning("Tried to reset animation in the server for a client controlled animator");
         return;
    }
    if (!hasAuthority) 
    {
         Debug.LogWarning("Only the client with authority can reset animations");
        return;
    }

    // ok,  do the Cmd
}
else
{
    //  check that this is only called on the server and issue warnings otherwise
}
```

Reply:
Separate PR for warnings